tests/drivers/build_all/ethernet/testcase.yaml

```diff
@@ -16,6 +16,15 @@ tests:
       - native_sim
       - native_sim/native/64
 
+  net.ethernet.build.spi.l2ethernet:
+    extra_args: DTC_OVERLAY_FILE="spi_devices_l2ethernet.overlay"
```

Comment:
I guess I'm a bit confused why you added a test case, what was the reason?

Reply:
Sorry for the delayed response.
I was worried that there would be cases where the settings would be exclusive with L2_ETHERNET, but I was overthinking it. I've combined them into one.